### DISCUSSION TITLE
LEGO original PID settings of powered up XL angular motor

### DISCUSSION THREAD

**Discussion by vanczakp** [pybricks/support#1526] [Q&A] [answered]
Hello! Does anybody know what is the original LEGO PID settings for angular XL motor? I am sure it is not the same compared to what comes with the pybricks firmware. My problem is: powered up the speed control is precise and smooth, but the pybrics settings (42484, 21242, 5310, 8, 15) results swinging - it is even worse if I reduce the max allowed speed (1200->800 and select slow acceleration (400).
I know it can be improved by trials but Im hoping the developer team know what was the original :)
Thanks in advance!

**Comment by laurensvalk** [comment]
The default values work better for most applications, but for large inertial loads like driving a truck you can make it smoother by increasing the profile value, as follows:

```
from pybricks.pupdevices import Motor, ColorDistanceSensor
from pybricks.parameters import Port, Direction, Stop
from pybricks.tools import wait

# Initialize the motors
steer = Motor(Port.C)

# Initialize the motors with increased smoothness profile.
front = Motor(Port.A, Direction.COUNTERCLOCKWISE, profile=360)
rear = Motor(Port.B, Direction.COUNTERCLOCKWISE, profile=360)
```

With the powered up app, the torque and precision are very much reduced, which is less ideal for robotics application (but better for cars where position is not important). Increasing the profile value achieves a similar effect.

**Reply by vanczakp** [reply]
Could you tell me what is the powered up parameters of PID? I know I was talking about the RS E-tron in regards of thisbefore but now Im working on the the liebherr crane. I would like to control everything by speed but regardless it is the main boom or the hoist without load, the PID loop results swinging as I described before.  I will try changing the PID parameters, the profile because I'm courious what is the effect but in this case the lego official settings going to be perfect as it was prooven for me before.

**Reply by laurensvalk** [reply, marked as answer]
It uses a completely different controller, so the PID values are not comparable.

Generally, the precision is much higher in Pybricks. Again, if you're using a heavy crane that is also heavily geared, precision is not important at all so you can decrease it to get smoother motion under heavy load.

Instead of trying numbers, you could read out the existing values and halve the Kp value, for example. That is what the profile will do automatically well if you prefer not to change the PID values.

**Reply by vanczakp** [reply]
thanks for the detailed explanation, well appreciated! Have a good day!

If anyone wondering what is the PID settings by pybrics, here it is:
        kp=42484  µNm/deg.
        ki=21242  µNm/(deg s).
        kd=5310  µNm/(deg/s).
        integral_deadzone=8
        integral_rate=15
control.pid(kp, ki, kd, integral_deadzone, integral_rate)
control.pid(42484, 21242, 5310, 8, 15)
